### ISSUE TITLE
How to set number of ccnsumers

### ISSUE THREAD

**Issue by OElesin** [relayr/rabbitmq-scala-client#5]
Great driver for rabbit MQ but how do I set number of consumers?

**Comment by platy** [comment]
The library doesn't do any consumer management at a higher level than the channel level.

You can pass your listener function to addConsumer on multiple channels or channels on multiple connections.

I'm not sure what the advantage of having the library managing multiple consumers would be. Do you have an example of another amqp library which does this?

**Comment by OElesin** [comment]
yeah. there's way to limit the consumers using actors and reactive programming

  var counter = 0
  def receive = {
    case PullData => {
      def consumer(request: Message): Unit = request match {
//        case Message.String(string) => BillingRequest.processQueue(((Json(DefaultFormats).parse(string)).values))
//        case _ => println(Message.apply(MessageProperties.apply(), ByteArray.apply()))
//        case Some() =>
        case Message.OctetStream(data) => println(data + " comes as octetstream");
        case Message.Array(data) => println(data. + " comes as an array");
        case _ => println("sorry we cant process at this time")
      }
      if(counter < 6){
        channel.addConsumer(thequeue, consumer)
        println("Log 1: Done pulling data " + counter)
        prompt ! AttemptPrompt
        counter += 1
      }

But how can I parse the values of Message.Array?

**Comment by platy** [comment]
Scala has no unapply for Array, so with `Message.Array` you can just put in an `Array[Byte]` and get out an `Array[Byte]`. The parsing is specific to the format of data in your message.

From your code it looks like you might be adding the same consumer function to the same queue up to 5 times, there probably are reasons for doing it but I haven't worked on any project where I wanted to use more than 1 consumer on a particular queue on a single channel.
